modules/agent-policy/scripts/script-utils.sh

```diff
@@ -0,0 +1,332 @@
+            "$instances_flag_name" "$instances_flag_value")"
+    fi
+
+    local command="gcloud $LAUNCH_STAGE compute instances ops-agents policies $action"
+    command="$command $policy_id$description_flag$agent_rules_flag$group_labels_flag"
+    command="$command$os_types_flag$zones_flag$instances_flag$update_flags"
+    echo "$command"
+}
+
+# Params:
+#   $1 = project id
+#   $2 = policy id
+#   $3 = description of the agent policy
+#   $4 = agent rules, in json format
+#   $5 = group labels, in json format
+#   $6 = os types, in json format
+#   $7 = zones, in json format
+#   $8 = instances, in json format
+# Return:
+#   the appropriate gcloud create command, given the args
+function get_create_command() {
+    local project_id="$1"
+    local policy_id="$2"
+    local description="$3"
+    local agent_rules_json="$4"
+    local group_labels_json="$5"
+    local os_types_json="$6"
+    local zones_json="$7"
+    local instances_json="$8"
+    local base_create_command
+    local global_flags
+
+    base_create_command="$(get_base_upsert_command "$CREATE" \
+        "$policy_id" "$description" "$agent_rules_json" \
+        "$group_labels_json" "$os_types_json" "$zones_json" "$instances_json")"
+    global_flags=$(get_global_flags "$project_id")
+
+    local create_command="$base_create_command$global_flags"
+    echo "$create_command"
+}
+
+# Params:
+#   $1 = project id
+#   $2 = policy id
+#   $3 = description of the agent policy
+#   $4 = agent rules, in json format
+#   $5 = group labels, in json format
+#   $6 = os types, in json format
+#   $7 = zones, in json format
+#   $8 = instances, in json format
+#   $9 = etag
+# Return:
+#   the appropriate gcloud update command, given the args
+function get_update_command() {
+    local project_id="$1"
+    local policy_id="$2"
+    local description="$3"
+    local agent_rules_json="$4"
+    local group_labels_json="$5"
+    local os_types_json="$6"
+    local zones_json="$7"
+    local instances_json="$8"
+    local etag="$9"
+    local base_update_command
+    local etag_flag
+    local global_flags
+
+    base_update_command="$(get_base_upsert_command "$UPDATE" \
+        "$policy_id" "$description" "$agent_rules_json" \
+        "$group_labels_json" "$os_types_json" "$zones_json" "$instances_json")"
+    etag_flag=$(get_flag etag "$etag")
+    global_flags=$(get_global_flags "$project_id")
+    local update_command="$base_update_command$etag_flag$global_flags"
+    echo "$update_command"
+}
+
+
+# Params:
+#   $1 = project id
+#   $2 = policy id
+# Return:
+#   the appropriate gcloud describe command, given the args
+function get_describe_command() {
+    local project_id="$1"
+    local policy_id="$2"
+    local project_flag_name="project"
+    local project_flag
+    project_flag=$(get_flag "$project_flag_name" "$project_id")
+
+    local command="gcloud $LAUNCH_STAGE compute instances ops-agents policies describe"
```

Comment:
Shouldn't we use json format for easier parsing? (Not a blocker.)

Reply:
That's a good idea! I'll make changes a bit later today.